src/test/java/com/microsoft/sqlserver/jdbc/bulkCopy/BulkCopyCSVTest.java

```diff
@@ -197,14 +200,20 @@ static void validateValuesFromCSV(DBTable destinationTable, String inputFile) {
                 ResultSetMetaData destMeta = ((ResultSet) dstResultSet.product()).getMetaData();
                 int totalColumns = destMeta.getColumnCount();
                 while (dstResultSet.next()) {
-                    String[] srcValues = br.readLine().split(delimiter);
-                    if ((0 == srcValues.length) && (srcValues.length != totalColumns)) {
+                    String[] srcValues = br.readLine().split(delimiter + "(?=([^\"]*\"[^\"]*\")*[^\"]*$)", -1);
+
+                    for (int i = 0; i < srcValues.length; i++) {
+                        srcValues[i] = srcValues[i].replaceAll("^\"|\"$", "").replaceAll("\"\"", "\"");
+                    }
+
+                    if ((0 == srcValues.length) || (srcValues.length != totalColumns)) {
                         srcValues = new String[totalColumns];
                         Arrays.fill(srcValues, null);
                     }
                     for (int i = 1; i <= totalColumns; i++) {
                         String srcValue = srcValues[i - 1];
                         String dstValue = dstResultSet.getString(i);
+                        srcValue = (srcValue.isEmpty()) ? null : srcValue;
```

Comment:
You need to do a null check for srcValue first. line 217 implies that srcValue can be null, which will result in NPE on this line upon calling isEmpty.